### PR TITLE
Story: add storylobby bool for friendly fire

### PR DESCRIPTION
**Pull request by 6fears7**

### CODE REVIEW

**Comment by 6fears7** [comment]
Need to find where we manage spear killing in story

**Comment by youbitchoc** [comment]
this should be synced for all players in the lobby, prolly gotta chuck it in storygamemode
i'd suggest to make it a checkbox in slugcatselectmenu

**Comment by 6fears7** [comment]
Sounds good. Did we have ff in non-msc at all? (inconsistent or not)

**Comment by youbitchoc** [comment]
> Need to find where we manage spear killing in story

Weapon.HitThisObject, hijack the `flag` bool

File: Story/StoryHooks.cs [hunk 1]
```diff
@@ -75,6 +75,21 @@ private void StoryHooks()
...
+        {
+            bool num = obj is Player && this is Spear && self.thrownBy != null && self.thrownBy is Player;
+            bool flag = (ModManager.CoopAvailable && RWCustom.Custom.rainWorld.options.friendlyFire) || (rainMeadowOptions.FriendlyFire.Value);
+            bool flag2 = self.room.game.IsArenaSession && self.room.game.GetArenaGameSession.arenaSitting.gameTypeSetup.spearsHitPlayers;
+            if (num)
+            {
+                return flag || flag2;
+            }
+
+            return true;
```
Comment (youbitchoc) [inline Story/StoryHooks.cs:92]: please don't rewrite the function :/ i can spin up an IL hook for ya

**Comment by 6fears7** [comment]
Could spare a lot of time by just adding an additional check in On. Instead of an IL for if story game mode and FF on, return true

**Comment by youbitchoc** [comment]
> Could spare a lot of time by just adding an additional check in On. Instead of an IL for if story game mode and FF on, return true

oh. yeah. we should do that

File: Story/StoryHooks.cs [hunk 2]
```diff
@@ -75,6 +77,48 @@ private void StoryHooks()
             On.SSOracleSwarmer.NewRoom += SSOracleSwarmer_NewRoom;
             On.HUD.TextPrompt.Update += TextPrompt_Update;
             On.HUD.TextPrompt.UpdateGameOverString += TextPrompt_UpdateGameOverString;
+
+            IL.Weapon.HitThisObject += Weapon_HitThisObject;
+        }
+
+        // bool flag2 = ModManager.CoopAvailable && Custom.rainWorld.options.friendlyFire;
+        // becomes
+        // bool flag2 = ModManager.CoopAvailable && Custom.rainWorld.options.friendlyFire || isStoryMode && rainmeadowOptions.FriendlyFire.Value;
```
[the remark this answers is not in the document]
Comment (youbitchoc) [reply Story/StoryHooks.cs:86]: ok yea just scrap this hook and return in On as you mentioned

**Pull request merged**